packages/admin-e2e-tests/src/pages/ProductsSetup.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import { waitForElementByText, getElementByText } from '../utils/actions';
+import { BasePage } from './BasePage';
+
+export class ProductsSetup extends BasePage {
+	url = 'wp-admin/admin.php?page=wc-admin&task=products';
+
+	async isDisplayed() {
+		await waitForElementByText( 'h1', 'Add products' );
+	}
+
+	async isStartWithATemplateDisplayed( templatesCount: number ) {
+		await waitForElementByText( 'h1', 'Start with a template' );
+		const length = await this.page.$$eval(
+			'.components-radio-control__input',
+			( items ) => items.length
+		);
+		expect( length === templatesCount ).toBeTruthy();
+	}
+
+	async clickStartWithTemplate() {
+		const startWithATemplate = await getElementByText(
+			'span',
+			'Start with a template'
+		);
+		await startWithATemplate?.click();
```

Comment:
You could make use of the `this.clickElementWithText` that you added to the `BasePage` right?
Also we might want to use a wildcard `*` instead of `span`

Reply:
Nice catch! I added those changes in commit 9ca1bdd